### ISSUE TITLE
How is everyone's VASA-1 doing?

### ISSUE THREAD

**Issue by xiaoxiaolai** [neeek2303/EMOPortraits#30]
https://github.com/user-attachments/assets/885d7df2-0094-42a8-aad9-d81b168d6a02

**Comment by zdyshine** [comment]
> 12.5_cxk_img.mp4

It looks okay. Is it trained according to the structure of VASA? I am also trying to train, but the training result is relatively poor. Can we communicate?

**Comment by xiaoxiaolai** [comment]
Of course, my results are also very poor.

**Comment by xiaoxiaolai** [comment]
> > 12.5_cxk_img.mp4
> 
> It looks okay. Is it trained according to the structure of VASA? I am also trying to train, but the training result is relatively poor. Can we communicate?看起来还不错。它是否根据 VASA 的结构进行训练？我也在尝试训练，但训练效果相对较差。我们能沟通吗？

Of course, my results are also very poor.

**Comment by xiaoxiaolai** [comment]
> > 12.5_cxk_img.mp4
> 
> It looks okay. Is it trained according to the structure of VASA? I am also trying to train, but the training result is relatively poor. Can we communicate?

Of course, my results are also very poor.

**Comment by zdyshine** [comment]
> > > 12.5_cxk_img.mp4
> > 
> > 
> > It looks okay. Is it trained according to the structure of VASA? I am also trying to train, but the training result is relatively poor. Can we communicate?看起来还不错。是按照VASA的结构来训练的吗？我也在尝试训练，但是训练效果比较差。我们可以交流吗？
> 
> Of course, my results are also very poor.当然，我的成绩也很差。

The training code is based on DiffPoseTalk?

**Comment by xiaoxiaolai** [comment]
yes

**Comment by zdyshine** [comment]
> yes

Me too,But the training effect is not very good either. I did it based on open source data. Yours is in Chinese. How long is the duration of your training data?

**Comment by zdyshine** [comment]
> yes

+方便QQ 781296632?

**Comment by xiaoxiaolai** [comment]
> > yes
> 
> +方便QQ 781296632?

微信吧
![WechatIMG808](https://github.com/user-attachments/assets/f4a912c4-32bb-4d4c-b570-d36c46a231e5)

**Comment by johndpope** [comment]
hi @xiaoxiaolai
 did you use cross attention between audio + motions? 
the paper says they use a transformer encoder (just self attention) - (as opposed to a transformer decoder - which has cross attention) 

when you encode the expression_embed - i'm struggling to make it stable and reproduce and converge. i feel like a code book of expressions maybe faster way to snapshot the latent space. im training 1 video - for thousand epochs - and while it does get better - it's like going to take years to converge. im put up the learning rate - and then it just blows up with inf / nans. did you just use standard learning rate? how bout gradient clipping? 
https://drive.google.com/drive/u/0/folders/1wiGA0g-QA0tKL-eHnNt5UJ97LVhekIxg

**Comment by xiaoxiaolai** [comment]
> hi [@xiaoxiaolai](https://github.com/xiaoxiaolai) did you use cross attention between audio + motions? the paper says they use a transformer encoder (just self attention) - (as opposed to a transformer decoder - which has cross attention)
> 
> when you encode the expression_embed - i'm struggling to make it stable and reproduce and converge. i feel like a code book of expressions maybe faster way to snapshot the latent space. im training 1 video - for thousand epochs - and while it does get better - it's like going to take years to converge. im put up the learning rate - and then it just blows up with inf / nans. did you just use standard learning rate? how bout gradient clipping? https://drive.google.com/drive/u/0/folders/1wiGA0g-QA0tKL-eHnNt5UJ97LVhekIxg

Maybe more data is needed for testing.
Diffusion models I referred https://github.com/DiffPoseTalk/DiffPoseTalk/
Do you use WeChat? You can add me

**Comment by Jason-Chi-xx** [comment]
@xiaoxiaolai hello, can we talk about the VASA training?  I've sent request to you in WeChat, Thanks

**Comment by johndpope** [comment]
if you train joyvasa (newer version of diffposetalk)- you will see similiar behaviour - they have expression embedding 70 dims / though its targeting liveportrait

https://github.com/jdh-algo/JoyVASA/pull/37

**Comment by johndpope** [comment]
i thought i was getting somewhere over last few weeks - but have hit a wall....

https://github.com/johndpope/VASA-1-hack/compare/feat/optimize-memory

https://wandb.ai/snoozie/vasa-overfitting/runs/79lp0h90?nw=nwusersnoozie
its not not built on diffposetalk - although i did cherry pick some joyvasa audio stuff 
- the vasa model is not validated.

i have some confusion on the model architecture - and not sure which direction is correct.
in the vasa paper - it mentions they built the transformer encoder - but from joyvasa - it's a transformer decoder - autoregressive (generate frame by frame attending to past frames only) - which i guess Microsoft mispoke ?

i built the vasamodel without the nonrigid warps - and trained model - then found the face was succesfully changing head angles - but no facial movements - the warps weren't wired up. so i threw out model and started again - and then opted to get vasa model to generate these warps (that's the successful generation in wandb) - but i had a conversation with grok this morning - and it's like - vasa shouldn't predict the warps it's too many - too unstable. 

but how did you guys do the vasa warps?  to generate the video from sound - did your vasa model predict the warps? 
the warps in this repo - needs a source + target - 

this code uses emo -> gets the warp -> saves it h5 
https://github.com/johndpope/VASA-1-hack/commit/4fb7904020294cf8c3e4748bfe8f940ed59bc042#diff-4bad4178fc8725099854fdcbb6fbc6538bd8d2eccb08167038d55d659132d6c8

my thinking was just dump these out of video - then get vasa to generate these 50x frame motions conditioned on audio
https://github.com/johndpope/VASA-1-hack/blob/feat/optimize-memory/vasa_model.py#L347 

there's some more notes here 
https://github.com/johndpope/VASA-1-hack/issues/49
 

-- UPDATE 

i rebuilt the model to derive warps from  expression - now training is converging. 
https://wandb.ai/snoozie/vasa-overfitting/runs/qqprlbs6/panel/b0gp2dkme?nw=nwusersnoozie
